### PR TITLE
fix some MKL detection issues of CMake

### PR DESCRIPTION
**Pull request by cyyever**
This PR rewrites some logic of FindMKL.cmake and FindOpenMP.cmake to better detect the corresponding libraries and fix the infinitely recursion between them. It also contains some other fixes without changing the CMake interface.

### CODE REVIEW

File: cmake/Modules/FindMKL.cmake [hunk 1]
```diff
@@ -41,10 +41,11 @@ IF (WIN32)
...
   SET(DEFAULT_INTEL_MKL_DIR "/opt/intel/mkl")
-  if (EXISTS "/opt/intel/oneapi")
-    SET(DEFAULT_INTEL_COMPILER_DIR "/opt/intel/oneapi")
-    if (EXISTS "/opt/intel/oneapi/mkl/latest")
-      SET(DEFAULT_INTEL_MKL_DIR "/opt/intel/oneapi/mkl/latest")
+  SET(DEFAULT_INTEL_ONEAPI_DIR "/opt/intel/oneapi")
+  if (EXISTS "${DEFAULT_INTEL_ONEAPI_DIR}")
+    SET(DEFAULT_INTEL_COMPILER_DIR "${DEFAULT_INTEL_ONEAPI_DIR}")
+    if (EXISTS "${DEFAULT_INTEL_ONEAPI_DIR}/mkl/latest")
+      SET(DEFAULT_INTEL_MKL_DIR "${DEFAULT_INTEL_ONEAPI_DIR}/mkl/latest")
```
Comment (malfet) [inline cmake/Modules/FindMKL.cmake:48]: Adding some explanation to PR description what this change is doing would be nice, imo.

**Review by malfet — APPROVED** [review]
LGTM, but please add some description about nature of those changes

File: cmake/Modules/FindMKL.cmake (hunk 1 above)
Comment (cyyever) [reply cmake/Modules/FindMKL.cmake:48]: I split the large commits into smaller ones and added comments.

**Review by Skylion007 — APPROVED** [review]

**Pull request closed without merging**